apps/sync/test/unit/adapters/history_test.js

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global
+  assert,
+  asyncStorage,
+  requireApp,
+  setup,
+  sinon,
+  suite,
+  teardown,
+  test,
+  SyncEngine
+*/
+
+var MockasyncStorage = {
+    mItems: {},
+
+    setItem: function(key, value, callback) {
+        this.mItems[key] = value;
+        if (typeof callback === 'function') {
+          callback();
+        }
+    },
+
+    getItem: function(key, callback) {
+      var value = this.mItems[key];
+      // use '|| null' will turn a 'false' to null
+      if (value === undefined) {
+        value = null;
+      }
+      if (typeof callback === 'function') {
+        callback(value);
+      }
+    },
+
+    mTeardown: function() {
+      this.mItems = {};
+    }
+};
+
+var MockIAC = {
+  request(channel, data) {
+    return Promise.resolve();
+  }
+};
+
+window.SyncEngine = {
+  DataAdapterClasses:{}
```

Comment:
Do this on the suite setup after saving the real SyncEngine and make sure you restore it on the suite teardown, please.